apps/api/src/donations/helpers/donation-status-updates.ts

```diff
@@ -35,6 +35,10 @@ export function shouldAllowStatusChange(
   oldStatus: DonationStatus,
   newStatus: DonationStatus,
 ): boolean {
+  if (oldStatus === newStatus) {
```

Comment:
I might be be missing something, but wouldn't short circuit make this check redundant ? What I mean is that whenever 
[donation?.status !== newDonationStatus ](https://github.com/podkrepi-bg/api/pull/436/commits/1f7c1c2dcab124771081edf9a9e023d7e12763a1#diff-88a7098916ffc30bd6ca323aa43535fbe9f8301dbc7a113af66c9b5bc8ec055bR477) evaluates to false, due to usage of  AND operator, shouldAllowStatusChange() shouldn't be called at all.

Reply:
That is true actually. Nice catch! That would not get called at all in that case. 
I think I will leave it though in the case this function is used somewhere else in the future. I think this check makes sense.

Reply:
@dimitur2204 yep this could be left here, however please check to remove the donation?.status !== newDonationStatus where the function is called